sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -163,9 +163,11 @@ object FunctionRegistry {
     expression[Abs]("abs"),
     expression[Coalesce]("coalesce"),
     expression[Explode]("explode"),
+    expressionGeneratorOuter[Explode]("explode_outer"),
```

Comment:
I am not super enthusiastic about this. We have three options here:

1. Leave as it is.
2. Remove the `outer_...` generators, and make a user use the `lateral view outer ...` instead.
3. Create separate OuterGenerator classes for each one, and provide proper documentation.

I am fine with any.